src/components/form/HdRange.vue

```diff
@@ -0,0 +1,274 @@
+<template>
+  <div class='range'>
+    <input type="range"
+      :name="name"
+      :v-model="value"
+      :required="required"
+      :disabled="disabled"
+      :min="minValue"
+      :max="maxValue"
+      :step="rangeStep"
+      @input="updateRangeDecoration"
+      @change="updateRangeDecoration"
+    >
+    <div class="range__thumb">
+      <div class="range__thumb__bullet">
+      </div>
+    </div>
+    <div class="range__decoration">
+      <div class="range__progress">
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+export default {
+  name: 'HdRange',
+  props: {
+    name: {
+      type: String,
+      default: 'range',
+    },
+    required: {
+      type: Boolean,
+      refault: false,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    minValue: {
+      type: Number,
+      default: 0,
+    },
+    maxValue: {
+      type: Number,
+      default: 100,
+    },
+    value: {
+      type: Number,
+      default: 0,
+    },
+    rangeStep: {
+      type: Number,
+      default: 1,
+    },
+  },
+  watch: {
+    value() {
+      this.input.value = this.value;
+      this.updateRangeDecoration();
+    },
+    minValue() {
+      this.updateRangeDecoration();
+    },
+    maxValue() {
+      this.updateRangeDecoration();
+    },
+  },
+  mounted() {
+    this.input = this.$el.querySelector('input');
+    this.progress = this.$el.querySelector('.range__progress');
+    this.progress.style.width = '1px';
+    this.thumb = this.$el.querySelector('.range__thumb');
+
+    if (this.minValue > this.maxValue) {
+      this.minValue = this.maxValue;
+      this.value = this.minValue;
+    }
+
+    this.updateRangeDecoration();
+  },
+  computed: {
+  },
+  methods: {
+    updateRangeDecoration() {
+      if (this.input.value > this.maxValue) {
+        this.input.value = this.maxValue;
```

Comment:
(Same this for the if statement bellow)

Instead of assigning to `this.input.value`, just emit an event `input` with the value you need, like so:
```js
this.$emit('input', this.maxValue);
```

v-model of the component (the one set on the parent) will take care of the binding :)

Reply:
Got it!